distributed/tests/test_client.py

```diff
@@ -6498,6 +6498,37 @@ def log_scheduler(dask_scheduler):
     assert events[1][1] == ("alice", "bob")
 
 
+@gen_cluster(client=True)
+async def test_log_event_warn(c, s, a, b):
+    def foo():
+        get_worker().log_event("warn", "Hello!")
+
+    with pytest.warns(Warning, match="Hello!"):
+        await c.submit(foo)
+
+
+@gen_cluster(client=True, Worker=Nanny)
+async def test_print(c, s, a, b, capsys):
+    def foo():
+        from dask.distributed import print
+
+        print("Hello!", 123, sep=":")
+
+    await c.submit(foo)
+
+    out, err = capsys.readouterr()
+    assert "Hello!:123" in out
+
+
+def test_print_simple(capsys):
+    from dask.distributed import print
```

Comment:
It's not clear to me if this `import` will impact other tests which are run later. I'm wondering if we should use

```python
from dask.distributed import print as dask_print
```

to not override the builtin `print` function for later tests

Reply:
I think that we're ok here.  I think that imports are properly scoped

```python
In [1]: def f():
   ...:     from dask.distributed import print
   ...: 

In [2]: print
Out[2]: <function print>

In [3]: print?
Docstring:
print(value, ..., sep=' ', end='\n', file=sys.stdout, flush=False)

Prints the values to a stream, or to sys.stdout by default.
Optional keyword arguments:
file:  a file-like object (stream); defaults to the current sys.stdout.
sep:   string inserted between values, default a space.
end:   string appended after the last value, default a newline.
flush: whether to forcibly flush the stream.
Type:      builtin_function_or_method
```

Reply:
Oops, I forgot to call `f() there, but I just did it again and it's fine.